### PR TITLE
Fix examples in API doc

### PR DESCRIPTION
**Pull request by lieser**

### CODE REVIEW

File: doc/api_ref/tls.rst [hunk 1]
```diff
@@ -601,7 +604,7 @@ The full code for a TLS server using asio is in `src/cli/tls_proxy.cpp`.
     class Server_Credentials : public Botan::Credentials_Manager
     {
        public:
-	  Server_Credentials() : m_key(Botan::PKCS8::load_key("botan.randombit.net.key"))
+          Server_Credentials() : m_key(Botan::PKCS8::load_key(Botan::DataSource_Stream("botan.randombit.net.key")))
```
Comment (lieser) [inline doc/api_ref/tls.rst:607]: In VS this still results in a warning:
```
warning C4239: nonstandard extension used: 'argument': conversion from 'Botan::DataSource_Stream' to 'Botan::DataSource &'
```

So may still not work on GCC/Clang. But did not see an easy fix without any bigger code changes. Any ideas?

File: doc/api_ref/tls.rst [hunk 2]
```diff
@@ -1531,7 +1534,7 @@ Server Code Example
     class Server_Credentials : public Botan::Credentials_Manager
     {
        public:
-          Server_Credentials() : m_key(Botan::PKCS8::load_key("botan.randombit.net.key")
+          Server_Credentials() : m_key(Botan::PKCS8::load_key(Botan::DataSource_Stream("botan.randombit.net.key")))
```
Comment (lieser) [inline doc/api_ref/tls.rst:1537]: Same C4239 warning here.

File: doc/api_ref/tls.rst (hunk 1 above)
Comment (lieser) [reply doc/api_ref/tls.rst]: Found a way to avoid the warning.

**Comment by reneme** [comment]
Did you find those in https://github.com/randombit/botan/pull/3025? If yes, I'd argue, we can close this and bring the fixes via the other PR.

**Comment by lieser** [comment]
Yes I did find them while working on #3025. I thought the fixes are unrelated enough to building them in the CI that it makes more sense to have a separate PR, and also didn't want to wait for the other PR to get this fixed.
But we can also do it on one PR if you prefer that.

**Review by reneme — APPROVED** [review]

**Pull request merged**